src/test/decompress_test.cpp

```diff
@@ -1,6 +1,45 @@
 #include "src/decompress.hpp"
 
+#include <boost/ut.hpp>
+
+#include <array>
+
 auto main() -> int
 {
-  return 0;
-}
+  using ::boost::ut::expect;
+  using ::boost::ut::test;
+
+  namespace huffman = ::starflate::huffman;
+  test("no compression block type") = [] {
+    constexpr auto compressed = std::array{
+        std::byte{0b10011111},
+        std::byte{5},
+        std::byte{0},  // len = 5
+        ~std::byte{5},
+        ~std::byte{0},  // nlen = 5
+        std::byte{'h'},
+        std::byte{'e'},
+        std::byte{'l'},
+        std::byte{'l'},
+        std::byte{'o'}};
+
+    const auto expected = std::vector{
+        std::byte{'h'},
+        std::byte{'e'},
+        std::byte{'l'},
+        std::byte{'l'},
+        std::byte{'o'}};
+
+    const auto actual =
+        starflate::decompress(std::span{compressed.data(), compressed.size()});
+    expect(::boost::ut::fatal(actual.has_value()))
+        << "got error code: " << static_cast<std::int32_t>(actual.error());
+    const auto& actual_value = actual.value();
+    expect(::boost::ut::fatal(actual_value.size() == expected.size()));
+    for (std::size_t i = 0; i < actual_value.size(); ++i) {
+      expect(actual_value[i] == expected[i])
+          << "at index " << i << "got " << static_cast<int>(actual_value[i])
+          << " but expected " << static_cast<int>(expected[i]);
+    }
```

Comment:
Yeah would be nice if the ut library did this stuff for us. Anyways got rid of it for now, just used it while debugging.

Reply:
I must have missed the notifications for this. I can try to write something that does this. It will speed up the debugging process which will we probably encounter again.

Reply:
```cpp
#include "test/ut.hpp"
#include <array>

auto main() -> int
{
  using ::boost::ut::expect;
  using ::boost::ut::test;
  using ::boost::ut::eq;

  using namespace ::boost::ut;

  test("test range eq") = [] {
    auto data1 = std::array{1, 2, 3};
    auto data2 = std::array{1, 2, 4};

    expect(eq(data1, data2));
  };
}
```

outputs this:
```sh
==================== Test output for //test:test:

Running test "test range eq"... FAILED
in: test/test.cpp:16 - test condition:  [{1, 2, 3} == {1, 2, 4}]


===============================================================================
Suite globaltests:   1 | 1 failed
asserts: 1 | 0 passed | 1 failed
```

Reply:
You can also do:
```cpp
using namespace ::boost::ut::operators;
expect(data1 == data2);
```